src/material/form-field/form-field.ts

```diff
@@ -685,4 +685,10 @@ export class MatFormField
     // shadow DOM, however browser that support shadow DOM should support `getRootNode` as well.
     return document.documentElement!.contains(element);
   }
+
+  /** Checks whether the form field is visible in DOM */
+  private _isVisibleInDom(): boolean {
+    const {width, height} = (this._elementRef.nativeElement as HTMLElement).getBoundingClientRect();
```

Comment:
I think a better fix for this might be to use `ResizeObserver` like we did for the outline notch (https://github.com/angular/components/pull/26028). Otherwise I'm worried about the performance implications of calling `getBoundingClientRect()` a bunch, as it will trigger re-layouts

Reply:
Thank you for your review. I have made some changes based on your feedback. Could you please take another look and let me know if you have any questions or concerns? I appreciate your time and input.